desktop-widgets/locationinformation.cpp

```diff
@@ -219,6 +220,21 @@ void LocationInformationWidget::initFields(dive_site *ds)
 	}
 }
 
+
+void LocationInformationWidget::on_GPSbutton_clicked()
+{
+	QString fileName = QFileDialog::getOpenFileName(this, tr("Select GPS file to open"),
+								tr("/home/willem/Desktop"),
```

Comment:
we shouldn't hardcode this directory. we should use the subsurface "application directory".

Reply:
You are absolutely correct. The application directory may differ between versions of subsurface. Specifically, in the Linux release versions (e.g. Ubuntu repository) the application directory may be in the system area of Linux (e.g. in the snap directory of Ubuntu). The safest option is probably the home directory. This is what I did.

Reply:
Another solution that is OS-independent would be to use the path provided by system_default_filename(). This would normally point to the directory containing the Subsurface settings.

Reply:
this should use `QString(system_default_directory())`
`/home` is not portable.

Reply:
I did exactly that in my latest commit.

Reply:
so I need to combine those commits. No problem.